test/integration/I18n_test.rb

```diff
@@ -176,4 +176,17 @@ class I18nTest < ActionDispatch::IntegrationTest
         assert true
       end
     end
+    
+    test "should choose i18n from http-accept-language" do
+      
+      # Must choose language from the accept language header on first-time load
+      get '/home', nil, {'HTTP_ACCEPT_LANGUAGE' => "de"}
+      assert_equal I18n.locale, :de
+      
+      # Must load language from cache after manual locale change
+      get_via_redirect "/change_locale/en"
+      assert_equal I18n.locale, :en
+      
+      assert true
```

Comment:
This is not needed, as it's a demo assertion that will never fail. Just remove this and this PR will be perfect, thanks! Good find on the `HTTP_ACCEPT_LANGUAGE` feature!